### PR TITLE
[RQ-1353] : Backgates restriction

### PR DESCRIPTION
**Pull request by RuntimeTerror10**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/features/rules/components/RulesList/components/RulesListIndex/components/ImportRulesModal/index.jsx [hunk 1]
```diff
@@ -209,14 +229,36 @@ export const ImportRulesModal = ({ toggle: toggleModal, isOpen }) => {
...
+      }
+
+      if (importType === RULE_IMPORT_TYPE.OVERWRITE || importType === RULE_IMPORT_TYPE.NORMAL) {
+        // by default overwrite is true so we dont need to process the importData again
+        doImportRules(importType);
+      } else {
+        processDataToImport(dataToImport, user, allRules, false);
+        doImportRules(importType);
+      }
+    },
```
Comment (nafees87n) [inline app/src/features/rules/components/RulesList/components/RulesListIndex/components/ImportRulesModal/index.jsx:258]:
```suggestion
      if (importType === RULE_IMPORT_TYPE.DUPLICATE) {        processDataToImport(dataToImport, user, allRules, false);
      } 
        doImportRules(importType);
    },
```

File: app/src/features/rules/constants/index.ts [hunk 2]
```diff
@@ -1,3 +1,9 @@
 import { RuleType } from "types";
 
 export const PREMIUM_RULE_TYPES = [RuleType.RESPONSE, RuleType.REQUEST, RuleType.SCRIPT];
+
+export const RULE_IMPORT_TYPE = {
+  OVERWRITE: "overwrite",
+  DUPLICATE: "duplicate",
+  NORMAL: "normal",
+};
```
Comment (nafees87n) [inline app/src/features/rules/constants/index.ts:9]: If default is overwrite then can we remove `normal`. Lets keep only 2 tyes - `overwrite` and `duplicate`

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx [hunk 3]
```diff
@@ -140,12 +142,16 @@ const useRuleTableColumns = (options: Record<string, boolean>) => {
...
-        // todo @nsr: check if this check also applies to groups?
         if (isRule(record)) {
           return (
             <PremiumFeature
               disabled={isRecordActive}
-              features={[FeatureLimitType.num_active_rules]}
+              features={
+                PREMIUM_RULE_TYPES.includes(record.ruleType)
+                  ? [FeatureLimitType.num_active_rules, FeatureLimitType.response_rule]
+                  : [FeatureLimitType.num_active_rules]
```
Comment (nafees87n) [inline app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx:152]: Can this check be done by PremiumFeatures component?
only ruleType to be passed and check if its a premium rule to be done by PremiumFeatures component

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx [hunk 4]
```diff
@@ -68,9 +68,11 @@ const useRuleTableActions = () => {
       const isRecordRule = isRule(record);
 
       //Push Notify
-      newStatus === RecordStatus.ACTIVE
-        ? toast.success(`${isRecordRule ? "Rule" : "Group"} is now ${newStatus.toLowerCase()}`)
-        : toast.success(`${isRecordRule ? "Rule" : "Group"} is now ${newStatus.toLowerCase()}`);
+      if (showToast) {
```
Comment (nafees87n) [inline app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx:71]: When is showToast optional?

File: app/src/components/features/rules/RuleBuilder/Header/ActionButtons/Status/index.js [hunk 5]
```diff
@@ -101,7 +103,12 @@ const Status = ({ isDisabled = false, location, isRuleEditorModal }) => {
             isRuleEditorModal ? "rule_editor_modal_header" : "rule_editor_screen_header"
           );
         }}
-        features={[FeatureLimitType.num_active_rules]}
+        features={
+          PREMIUM_RULE_TYPES.includes(currentlySelectedRuleData.ruleType)
+            ? [FeatureLimitType.num_active_rules, FeatureLimitType.response_rule]
+            : [FeatureLimitType.num_active_rules]
```
Comment (nafees87n) [inline app/src/components/features/rules/RuleBuilder/Header/ActionButtons/Status/index.js:109]: Check the other comment to only pass ruleType

**Review by nafees87n — CHANGES_REQUESTED** [review]

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx (hunk 4 above)
Comment (RuntimeTerror10) [reply app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx:71]: For free plan users ,to toggle the status to off of premium rules when table mounts.

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx (hunk 3 above)
Comment (RuntimeTerror10) [reply app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx:152]: PremiumFeatures component is a HOC which wraps multiple premium features touchpoints like dynamic request/response body, rule sharing etc.We shouldn't write any feature specific logic inside this HOC

**Review by nafees87n — APPROVED** [review]

**Pull request merged**